### DISCUSSION TITLE
Query parameters that are only needed client-side

### DISCUSSION THREAD

**Discussion by BrentChesny** [inertiajs/inertia#617] [Help (React)]
I am working on an Inertia-react application that has a react component that can be manipulated to explore some data. I want to encode some state of this component as url query parameters, so that a specific view can be shared by sharing the url. What would be the best way to update the query parameters in an inertia compliant way?

- Using `window.history.replaceState()` doesn't seem to work when I navigate away and then go back.
- Using `Inertia.replace()` would make unnecessary server calls I guess?

Any insights would be greatly appreciated 🙂

**Comment by stefanocurnis** [comment]
I have this problem too. Did you ever find a solution?

**Comment by Ivan-WG** [comment]
the better approach is to not use Inertia::Render, but with API calls to manage the state
